ethereum/jsonrpc/src/main/java/tech/pegasys/pantheon/ethereum/jsonrpc/internal/methods/privacy/eea/EeaGetTransactionCount.java

```diff
@@ -0,0 +1,138 @@
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcResponse;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.response.JsonRpcSuccessResponse;
+import tech.pegasys.pantheon.ethereum.jsonrpc.internal.results.Quantity;
+import tech.pegasys.pantheon.ethereum.privacy.PrivateTransactionHandler;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.collect.Lists;
+import org.apache.logging.log4j.Logger;
+import org.bouncycastle.util.Arrays;
+
+public class EeaGetTransactionCount implements JsonRpcMethod {
+
+  public static class EeaGetTransactionCountJsonBody {
+
+    final String privateFrom;
+    final String[] privateFor;
+    final Address address;
+
+    @JsonCreator
+    public EeaGetTransactionCountJsonBody(
+        @JsonProperty("privateFrom") final String privateFrom,
+        @JsonProperty("privateFor") final String[] privateFor,
+        @JsonProperty("address") final Address address) {
+      this.privateFrom = privateFrom;
+      this.privateFor = privateFor;
+      this.address = address;
+    }
+
+    public String getPrivateFrom() {
+      return privateFrom;
+    }
+
+    public String[] getPrivateFor() {
+      return privateFor;
+    }
+
+    public Address getAddress() {
+      return address;
+    }
+  }
+
+  private static final Logger LOG = getLogger();
+
+  private final JsonRpcParameter parameters;
+  private final Enclave enclave;
+  private final PrivateTransactionHandler privateTransactionHandler;
+
+  public EeaGetTransactionCount(
+      final JsonRpcParameter parameters,
+      final Enclave enclave,
+      final PrivateTransactionHandler privateTransactionHandler) {
+    this.parameters = parameters;
+    this.enclave = enclave;
+    this.privateTransactionHandler = privateTransactionHandler;
+  }
+
+  @Override
+  public String getName() {
+    return RpcMethod.PRIV_GET_LEGACY_TRANSACTION_COUNT.getMethodName();
+  }
+
+  @Override
+  public JsonRpcResponse response(final JsonRpcRequest request) {
+    if (request.getParamLength() != 1) {
+      return new JsonRpcErrorResponse(request.getId(), JsonRpcError.INVALID_PARAMS);
+    }
+    final EeaGetTransactionCountJsonBody eeaGetTransactionCount =
+        parameters.required(request.getParams(), 0, EeaGetTransactionCountJsonBody.class);
+
+    try {
+      final long nonce = determineNonce(eeaGetTransactionCount);
+      return new JsonRpcSuccessResponse(request.getId(), Quantity.create(nonce));
+    } catch (final Exception e) {
+      LOG.error("Failed to fetch group from Enclave with error " + e.getMessage());
+      LOG.error(e);
+      return new JsonRpcErrorResponse(request.getId(), JsonRpcError.FIND_PRIVACY_GROUP_ERROR);
+    }
+  }
+
+  private long determineNonce(final EeaGetTransactionCountJsonBody getTransactionCount)
+      throws Exception {
+
+    final String[] groupMembers =
```

Comment:
Perhaps there should be a method on the PrivateTransactionHandler to get the legacy privacy group?

Reply:
I couldn't decide if this is something a PrivateTransactionHandler should know about ...